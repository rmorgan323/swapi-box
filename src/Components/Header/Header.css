.header-component {
  width: 1273px;
  margin: auto; }
  .header-component h1 {
    color: #fff;
    opacity: .8;
    text-align: center; }
  .header-component .logo {
    background-image: url("swapi-logo.svg");
    width: 300px;
    height: 162px;
    background-repeat: no-repeat;
    margin: 0 auto 19px; }
  .header-component .header-buttons {
    display: flex;
    justify-content: space-around;
    margin-bottom: 9px; }
  .header-component button {
    border: none;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: rgba(42, 252, 255, 0.65);
    height: 40px;
    width: 220px;
    font-family: 'News Gothic Std', sans-serif;
    font-weight: 900;
    position: relative;
    font-size: 20px;
    color: #fff;
    overflow: hidden;
    cursor: pointer;
    padding-top: 4px;
    border-radius: 20px; }
  .header-component button:hover {
    background-color: rgba(42, 252, 255, 0.8);
    box-shadow: 0px 0px 56px -6px #dbe2ff; }
  .header-component span {
    position: absolute;
    right: 0;
    top: -5px;
    background-color: rgba(0, 93, 94, 0.5);
    padding: 14px; }

@media screen and (max-width: 1272px) {
  .header-component {
    width: initial; } }

@media screen and (max-width: 925px) {
  .header-component .header-buttons {
    margin-bottom: -5px; }
  .header-buttons {
    flex-direction: column;
    align-items: center; }
    .header-buttons button {
      margin-bottom: 13px; } }
