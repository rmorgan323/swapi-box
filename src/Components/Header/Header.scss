.header-component {
	margin: auto;
	width: 1273px;

	h1 {
		color: #fff;
		opacity: .8;
		text-align: center;
	}

	.logo {
		background-image: url('swapi-logo.svg');
		background-repeat: no-repeat;
		height: 162px;
		margin: 0 auto 19px;
		width: 300px;
	}

	.header-buttons {
		display: flex;
		justify-content: space-around;
		margin-bottom: 9px;
	}

	button {
		align-items: center;
		background-color: rgba(42,252,255, .65);
		border: none;
		border-radius: 20px;
		color: #fff;
		cursor: pointer;
		display: flex;
		font-family: 'News Gothic Std', sans-serif;
		font-size: 20px;
		font-weight: 900;
		height: 40px;
		justify-content: center;
		overflow: hidden;
		padding-top: 4px;
		position: relative;
		width: 220px;
		z-index: 1;
	}

	button:hover {
		background-color: rgba(42,252,255, .8);
		box-shadow: 0px 0px 56px -6px rgba(219,226,255,1);
	}

	span {
		background-color: rgba(0,93,94, .5);
		padding: 14px;
		position: absolute;
		right: 0;
		top: -5px;
	}
}

@media screen and (max-width: 1272px) {
	.header-component {
		width: initial;
	}
}

@media screen and (max-width: 925px) {
	.header-component {
		.header-buttons {
			margin-bottom: -5px;
		}
	}
	.header-buttons {
		align-items: center;
		flex-direction: column;

		button{
			margin-bottom: 13px;
		}
	}
}