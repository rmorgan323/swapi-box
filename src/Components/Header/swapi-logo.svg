<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 21.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 590.1 318.5" style="enable-background:new 0 0 590.1 318.5;" xml:space="preserve">
<style type="text/css">
	.st0{opacity:0.8;}
	.st1{fill:url(#SVGID_1_);}
	.st2{fill:url(#SVGID_2_);}
	.st3{fill:url(#SVGID_3_);}
	.st4{fill:url(#SVGID_4_);}
	.st5{fill:url(#SVGID_5_);}
</style>
<g class="st0">
	<linearGradient id="SVGID_1_" gradientUnits="userSpaceOnUse" x1="280.415" y1="150.4236" x2="476.7659" y2="58.8637">
		<stop  offset="0" style="stop-color:#FFFFFF;stop-opacity:0.78"/>
		<stop  offset="1" style="stop-color:#FFFFFF;stop-opacity:0.6"/>
	</linearGradient>
	<path class="st1" d="M428.3,124.3h30.2c8-1.2,14.4-3.8,19.2-7.9c4.8-4.1,7.7-7.7,8.8-10.8c1.1-3.1,1.8-6.5,2.1-10
		c0.4-3.6-0.4-8.9-2.4-16c-2.1-5.9-5.7-10.5-10.8-13.7c-5.1-3.3-11.5-5.3-19.2-6h-62.7v92.4l-30.9-92.4h-47.8l-31.7,96.4h34.6
		l6.1-17.3h28.6l6.1,17.3h35.1h1.3h33.5V124.3z M428.3,83.8h24.2c1.1,0,2.4,0.6,4,1.8c1.6,1.2,2.5,3.6,2.7,7.3
		c-0.1,3.9-0.9,6.3-2.4,7.2c-1.5,1-2.9,1.4-4.2,1.4h-24.3V83.8z M325,119.3l13.8-38.2l12.7,38.2H325z"/>
	<linearGradient id="SVGID_2_" gradientUnits="userSpaceOnUse" x1="476.6772" y1="122.5387" x2="539.3458" y2="93.3158">
		<stop  offset="0" style="stop-color:#FFFFFF;stop-opacity:0.6"/>
		<stop  offset="1" style="stop-color:#FFFFFF;stop-opacity:0.56"/>
	</linearGradient>
	<rect x="492.3" y="59.7" class="st2" width="31.4" height="96.4"/>
	<linearGradient id="SVGID_3_" gradientUnits="userSpaceOnUse" x1="189.5073" y1="237.6237" x2="290.6871" y2="190.4428">
		<stop  offset="0" style="stop-color:#FFFFFF;stop-opacity:0.75"/>
		<stop  offset="1" style="stop-color:#FFFFFF;stop-opacity:0.65"/>
	</linearGradient>
	<path class="st3" d="M239.9,164.6c-8.7,0-16.4,1.1-23.3,3.2c-6.9,2.2-12.6,5.3-17.3,9.5c-4.7,4.2-8.3,9.3-10.7,15.4
		c-2.4,6.1-3.7,13.2-3.7,21.3c0,10.6,2.1,19.6,6.4,27c4.3,7.4,10.6,12.9,18.9,16.7c8.3,3.8,18.2,5.6,29.8,5.6
		c11.6,0,21.5-1.9,29.9-5.6c8.3-3.7,14.7-9.3,19.1-16.7c4.4-7.4,6.6-16.4,6.6-27c0-10.5-2.2-19.5-6.6-26.9
		c-4.4-7.4-10.8-13-19.2-16.8C261.4,166.5,251.5,164.6,239.9,164.6z M259.8,225.5c-0.8,3.1-2.7,5.8-5.9,8.2s-7.9,3.6-14.2,3.6
		c-8.2,0-13.7-2.3-16.4-6.9c-2.8-4.6-4.2-10-4.2-16.1c0-15.4,6.9-23.1,20.6-23.1c8.7,0,14.4,2.3,17.1,6.9c2.7,4.6,4.2,10,4.5,16.2
		C261.1,218.7,260.6,222.4,259.8,225.5z"/>
	<linearGradient id="SVGID_4_" gradientUnits="userSpaceOnUse" x1="279.3256" y1="244.2915" x2="411.1666" y2="182.813">
		<stop  offset="0" style="stop-color:#FFFFFF;stop-opacity:0.65"/>
		<stop  offset="1" style="stop-color:#FFFFFF;stop-opacity:0.55"/>
	</linearGradient>
	<polygon class="st4" points="365,165.4 345.1,195.1 325.5,165.4 287.5,165.4 322.8,213.6 287.5,261.7 325.5,261.7 345.1,232 
		365,261.7 403,261.7 367.7,213.6 403,165.4 	"/>
	<linearGradient id="SVGID_5_" gradientUnits="userSpaceOnUse" x1="-8.3069" y1="17.8142" x2="598.3821" y2="300.718">
		<stop  offset="0" style="stop-color:#FFFFFF"/>
		<stop  offset="1" style="stop-color:#FFFFFF;stop-opacity:0.5"/>
	</linearGradient>
	<path class="st5" d="M0,0v156.2h77.6v-0.1h66.2c13.9-2.3,21-9.4,21.2-21.4c0.3-11.9-6.8-24.2-21.2-36.8c-8.1-6.7-6.4-9.8,5.1-9.5
		h18.8l20.6,67.7h31.4l10.1-40.8l12.2,40.8h31.4l24.2-96.4h-31.4l-8.9,41.6l-13.4-41.6H215l-11.5,41.6l-13.2-41.6h-11.1h-20.2h-39.2
		c-12.3,0-19.6,6-21.9,17.9c-2.3,11.9,4.3,25,19.8,39.2c9.4,7.4,7.8,10.9-4.8,10.5H59.7v0.1h-31V28.7h532.7v261.1H28.7V194h30.9v0.1
		h20v67.7h60.8c8.7,0.1,16.1-0.5,22.3-1.7c6.1-1.2,11.1-3.8,15-7.7c3.9-3.9,5.8-9.7,5.8-17.3c0-2.4-0.6-4.8-1.7-7.3
		c-1.1-2.5-2.7-4.7-4.7-6.7c-2-2-4.5-3.7-7.5-5.1c-3-1.4-6.2-2.4-9.7-2.8v-0.3c7.2-2.4,12.3-5.4,15.4-9c3.1-3.6,4.5-8.7,4.1-15.1
		c-1.2-8.4-4.5-14.3-10-17.9c-5.5-3.6-13.5-5.4-24.1-5.5H77.6v-0.1H0v153.2h590.1V0H0z M112.8,188.7h23.3c1,0,2.3,0.5,3.8,1.6
		c1.5,1.1,2.4,2.9,2.6,5.5c-0.1,2.3-0.8,4.1-2.3,5.4c-1.4,1.4-2.8,2-4.1,2h-23.4V188.7z M112.8,224.4h21.4c4.4,0,7.9,0.8,10.4,2.4
		c2.5,1.6,3.6,3.8,3.4,6.6c0,2.5-1.4,4.7-4.1,6.4c-2.8,1.7-6,2.5-9.7,2.4h-21.4V224.4z"/>
</g>
</svg>
