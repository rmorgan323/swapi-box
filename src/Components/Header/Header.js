import React from 'react';
import './Header.css';

const Header = (props) => {

	return (
		<div className="header-component">
			{console.log('header component', props)}
			<h1>swAPIbox</h1>
			<h6>FAVORITES <span>{props.favorites.length}</span></h6>
			<div>
				<button onClick={props.getCharacters}>characters</button>
				<button onClick={props.getWorlds}>worlds</button>
				<button onClick={props.getVehicles}>vehicles</button>
			</div>
		</div>
	)
}




export default Header;