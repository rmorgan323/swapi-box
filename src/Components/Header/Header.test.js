import React from 'react';
import { shallow, mount } from 'enzyme';
import Header from './Header';

describe('Header test', () => {
  it('should render correctly', () => {
    const renderedHeader = shallow(<Header />);

    expect(renderedHeader.find('.header-component').length).toEqual(1);
  });

  it('should receive props', () => {
    const renderedHeader = mount(<Header numberOfFavorites={8} />);

    expect(renderedHeader.props().numberOfFavorites).toEqual(8);
  });

  it('should match the snapshot', () => {
    const renderedHeader = shallow(<Header numberOfFavorites={5} />);

    expect(renderedHeader).toMatchSnapshot();
  });
});