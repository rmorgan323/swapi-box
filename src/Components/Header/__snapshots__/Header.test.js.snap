// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    numberOfFavorites={5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo"
/>,
        <div
          className="header-buttons"
>
          <button
                    onClick={[Function]}
          >
                    CHARACTERS
          </button>
          <button
                    onClick={[Function]}
          >
                    WORLDS
          </button>
          <button
                    onClick={[Function]}
          >
                    VEHICLES
          </button>
          <button
                    onClick={[Function]}
          >
                    FAVORITES
                    <span
                              className="faves"
                    >
                              5
                    </span>
          </button>
</div>,
      ],
      "className": "header-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "logo",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
>
              CHARACTERS
</button>,
            <button
              onClick={[Function]}
>
              WORLDS
</button>,
            <button
              onClick={[Function]}
>
              VEHICLES
</button>,
            <button
              onClick={[Function]}
>
              FAVORITES
              <span
                            className="faves"
              >
                            5
              </span>
</button>,
          ],
          "className": "header-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CHARACTERS",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "CHARACTERS",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "WORLDS",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "WORLDS",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "VEHICLES",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "VEHICLES",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "FAVORITES",
                <span
                  className="faves"
>
                  5
</span>,
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "FAVORITES",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 5,
                  "className": "faves",
                },
                "ref": null,
                "rendered": 5,
                "type": "span",
              },
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo"
/>,
          <div
            className="header-buttons"
>
            <button
                        onClick={[Function]}
            >
                        CHARACTERS
            </button>
            <button
                        onClick={[Function]}
            >
                        WORLDS
            </button>
            <button
                        onClick={[Function]}
            >
                        VEHICLES
            </button>
            <button
                        onClick={[Function]}
            >
                        FAVORITES
                        <span
                                    className="faves"
                        >
                                    5
                        </span>
            </button>
</div>,
        ],
        "className": "header-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "logo",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
>
                CHARACTERS
</button>,
              <button
                onClick={[Function]}
>
                WORLDS
</button>,
              <button
                onClick={[Function]}
>
                VEHICLES
</button>,
              <button
                onClick={[Function]}
>
                FAVORITES
                <span
                                className="faves"
                >
                                5
                </span>
</button>,
            ],
            "className": "header-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CHARACTERS",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "CHARACTERS",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "WORLDS",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "WORLDS",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "VEHICLES",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "VEHICLES",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "FAVORITES",
                  <span
                    className="faves"
>
                    5
</span>,
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "FAVORITES",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 5,
                    "className": "faves",
                  },
                  "ref": null,
                  "rendered": 5,
                  "type": "span",
                },
              ],
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
